File: Cryptofolio/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Cryptofolio.Models
{
    public class Transaction
    {

        public Transaction() { }

        [Key]
        public int Id { get; set; }

        public DateTime Date { get; set; }

        public string Note { get; set; } = string.Empty;

        public float Amount { get; set; }

        public float Fee { get; set; }

        public string ApplicationUserId { get; set; } = String.Empty;


        [ForeignKey ("ApplicationUserId")]
        public ApplicationUser? ApplicationUser { get; set; }

        public string CoinSymbol { get; set; } = string.Empty;

        [ForeignKey("CoinSymbol")]
        public Coin? Coin { get; set; }

        public ICollection<FinanceTransaction> FinanceTransactions { get; set; }

        public ICollection<TransferTransaction> TransferTransactions { get; set; }



    }
}

File: Cryptofolio/ClientApp/src/app/common/models/transaction-models/transaction.ts
import { DateTime } from "@syncfusion/ej2-angular-charts";

export class Transaction{

  public ['@odata.type'] = '';

  public Id : number = -1;

  public Date: Date = new Date;

  public Note : string = "";

  public Amount : number = 0;

  public Fee : number = 0;

  public ApplicationUserId : string = "";

  public CoinSymbol : string = "";

  // public Price : number = 0;

  constructor(data?: Partial<Transaction>) {
    Object.assign(this, data);
  }
}
